prime-router/docs/authz/authz-deprecated-implementation.md

```diff
@@ -0,0 +1,151 @@
+# Authorization Deep Dive
+
+## Current Design
+
+To authorize a user for an endpoint, we compare the scopes that are present on the access token versus what scope is 
+required.
+
+### Scope Design
+
+- A client can have multiple scopes
+- Scopes are in the format of `{organization}.{senderOrReceiver}.{role}` 
+- `organization` and `senderOrReceiver` can be omitted with a wildcard `*` or `default`
+
+There are four possible roles. 
+
+| Role       | Definition                                                                |
+|------------|---------------------------------------------------------------------------|
+| primeadmin | system administrator with full access to all endpoints                    |
+| admin      | administrator with access to endpoints that manage their own organization |
+| user       | user with read access to endpoints scoped to their organization           |
+| report     | able to submit a report as an organization                                |
+
+
+Here are some examples given an organization of md-phd (Maryland Public Health Department)
+
+| Scope                   | Access                                                                 |
+|-------------------------|------------------------------------------------------------------------|
+| `*.*.primeadmin`        | System Administrator                                                   |
+| `md-phd.*.admin`        | Organization Administrator for md-phd                                  |
+| `md-phd.*.user`         | Organization User for md-phd                                           |
+| `md-phd.*.report`       | Organization Report Submitter for md-phd                               |
+| `md-phd.default.admin`  | Organization Administrator for md-phd                                  |
+| `md-phd.default.user`   | Organization User for md-phd                                           |
+| `md-phd.default.report` | Organization Report Submitter for md-phd                               |
```

Comment:
AFAIK, there is functionally no difference between 'default' and 'abc'. 'default' is nothing special and literally means "a sender/receiver named 'default'" and 'abc' means "a sender/receiver named 'abc'"

Reply:
`default` is actually a special case and is treated equivalently to `*`.

https://github.com/CDCgov/prime-reportstream/blob/main/prime-router/src/main/kotlin/tokens/AuthenticatedClaims.kt#L124